tests/Aspire.Workload.Tests/AppHostTemplateTests.cs

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+using Xunit.Abstractions;
+using System.Text.RegularExpressions;
+
+namespace Aspire.Workload.Tests;
+
+public partial class AppHostTemplateTests : WorkloadTestsBase
+{
+    public AppHostTemplateTests(ITestOutputHelper testOutput)
+        : base(testOutput)
+    {
+    }
+
+    [Fact]
+    public async Task EnsureProjectsReferencing8_1_0WorkloadCanBuild()
+    {
+        string projectId = "aspire-can-reference-8.1.0";
+        await using var project = await AspireProject.CreateNewTemplateProjectAsync(
+            projectId,
+            "aspire-apphost",
```

Comment:
Would it be useful to test with the starter template instead, and check that it runs as expected? We have existing code for that.

Reply:
I saw we weren't testing this template so thought about using it too to get coverage, plus this particular test really only cares about building, as if this scenario is regressed the main issue would be that the user would not be able to build.

Reply:
Yeah, we need to add tests for `apphost`, and the `servicedefaults` project (https://github.com/dotnet/aspire/issues/2689).